skyportal/handlers/api/webhook.py

```diff
@@ -70,63 +65,67 @@ def post(self, analysis_resource_type, token):
-                    f"Analysis already updated with status='{analysis.status}'"
-                    f" and message={analysis.status_message}",
-                    status=403,
+
+        with self.Session() as session:
+            try:
+                analysis = (
+                    session.query(ObjAnalysis)
+                    .filter(ObjAnalysis.token == token)
+                    .first()
                 )
-            if (
-                analysis.invalid_after
-                and datetime.datetime.utcnow() > analysis.invalid_after
-            ):
-                analysis.status = 'timed_out'
-                analysis.status_message = f'Analysis timed out before webhook call at {str(datetime.datetime.utcnow())}'
+                if not analysis:
+                    return self.error("Invalid token", status=403)
+                last_active = analysis.last_activity
+                if analysis.status not in ['pending', 'queued']:
+                    return self.error(
+                        f"Analysis already updated with status='{analysis.status}'"
+                        f" and message={analysis.status_message}",
+                        status=403,
+                    )
+                if (
+                    analysis.invalid_after
+                    and datetime.datetime.utcnow() > analysis.invalid_after
+                ):
+                    analysis.status = 'timed_out'
+                    analysis.status_message = f'Analysis timed out before webhook call at {str(datetime.datetime.utcnow())}'
+                    analysis.last_activity = datetime.datetime.utcnow()
+                    analysis.duration = (
+                        analysis.last_activity - last_active
+                    ).total_seconds()
+                    session.commit()
+                    session.close()
+                    return self.error("Token has expired", status=400)
+
+                # lock the analysis associated with this token and commit immediately to avoid race conditions,
+                # so that the results are not written more than once
+                analysis.status = 'completed'
                 analysis.last_activity = datetime.datetime.utcnow()
                 analysis.duration = (
                     analysis.last_activity - last_active
                 ).total_seconds()
                 session.commit()
-                session.close()
-                return self.error("Token has expired", status=400)
-
-            # lock the analysis associated with this token and commit immediately to avoid race conditions,
-            # so that the results are not written more than once
-            analysis.status = 'completed'
-            analysis.last_activity = datetime.datetime.utcnow()
-            analysis.duration = (analysis.last_activity - last_active).total_seconds()
+            except Exception as e:
+                log(f'Troubling accessing Analysis with token {token} {e}.')
```

Comment:
Is this the most likely error and result for this exception? Aren't we checking for tokens above?

Reply:
If I try to interpret what the original author of this error message meant, I would probably guess the idea was "here's an error, here's the token (so you'd have some context) and here's the error message itself. 
Perhaps a better way to make that clearer would be `Trouble posting Analysis results. Token: {token}. Error: {e}`.

Reply:
@guynir42 agreed, that seems better to me.

Reply:
I am wondering if we should be returning more info in the `self.error`. Could it be that returning the token/error message reveals internals that we don't necessarily want to expose?

Reply:
Presumably it is the token that the user sent, so seems useful to spit it back.

Reply:
how about now?

Reply:
Ok by me! Good to merge on green!